docs/further_useful_features.rst

```diff
@@ -0,0 +1,432 @@
+.. _further_useful_features:
+
+=================================
+Further useful kartothek features
+=================================
+
+
+This document introduces additional features of ``kartothek`` that allow users to
+partition and index their data during writing, update already written data and
+keep their data stores 'clean' with garbage collection.
+
+The intent of this document is to expose users to some additional features of ``kartothek``
+in a bid to encourage deeper exploration of ``kartothek`` beyond 'getting started' and help
+better enable developers to start thinking about how it can (or cannot) be useful in their
+particular applications.
+
+
+.. _partitioning_section:
+
+Partitioning and Indexing
+=========================
+
+``kartothek`` is designed primarily for storing large datasets consistently and
+accessing them efficiently. To achieve this, it provides two useful functionalities:
+partitioning and secondary indices.
+
+
+Partitioning
+------------
+
+As we have already seen, updating a dataset in ``kartothek`` amounts to adding new
+partitions, which in the underlying key-value store translates to writing new files
+to the storage layer.
+
+From the perspective of efficient access, it would be helpful if accessing a subset
+of written data didn't require reading through an entire dataset to be able to
+identify and access the required subset. This is where *explicitly* partitioning by
+table columns helps.
+
+To see explicit partitioning in action, let's set up some data and a storage location
+first and store the data there with ``kartothek``:
+
+.. ipython:: python
+
+    import numpy as np
+    import pandas as pd
+    from functools import partial
+    from tempfile import TemporaryDirectory
+    from storefact import get_store_from_url
+
+    from kartothek.io.eager import store_dataframes_as_dataset
+
+    dataset_dir = TemporaryDirectory()
+
+    store_factory = partial(get_store_from_url, f"hfs://{dataset_dir.name}")
```

Comment:
can we somewhere here mentioned that we require extended keyspace stores? See #69 for details.